src/main/java/com/splunk/kafka/connect/SplunkSinkConnector.java

```diff
@@ -66,4 +80,105 @@ public String version() {
     public ConfigDef config() {
         return SplunkSinkConnectorConfig.conf();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues =
+            config.configValues()
+                .stream()
+                .collect(Collectors.toMap(
+                    ConfigValue::name,
+                    Function.identity()));
+
+        SplunkSinkConnectorConfig connectorConfig;
+        try {
+            connectorConfig = new SplunkSinkConnectorConfig(connectorConfigs);
+        } catch (Exception e) {
+            log.error("Validating configuration caught an exception", e);
+            return config;
+        }
+
+        validateAccess(connectorConfig, configValues);
+
+        return config;
+    }
+
+    /**
+     * We validate access by posting an empty payload to the Splunk endpoint.
+     *
+     * For a valid endpoint and a valid token, this returns a HTTP 400 Bad Request with the
+     * payload: {"text":"No data","code":5}
+     * For a valid endpoint and an invalid token, this returns a HTTP 403 Forbidden with the
+     * payload: {"text":"Invalid token","code":4}
+     * For an invalid hostname and other errors, the Java UnknownHostException and other similar
+     * Exceptions are thrown.
+     *
+     * @param connectorConfig The connector configuration
+     * @param configValues The configuration ConfigValues
+     */
+    private void validateAccess(SplunkSinkConnectorConfig connectorConfig, Map<String, ConfigValue> configValues) {
```

Comment:
I feel this function is slightly overloaded. While I believe the validations are a really good way to give users feedback it's also really imp. to keep them simple IMO. Sending back error messages from the exceptions or from API responses can unknowingly bite us back as we can't keep track of the contract. 

On the flip side, we do understand what the HTTP responses really mean as they are universally accepted contracts. I would agree with @aniketshrimal too here (https://github.com/confluentinc/kafka-connect-splunk/pull/5/files#r756709867) that we should be replacing them with our own messages.

It might even be alright to fail fast and record simple errors even if one of the HEC connections fails.

Reply:
The error space is vast: It isn't just the HTTP error codes, it's also the Exceptions such as Unknown host, or failed to connect to the port and so on.

It would not be possible to enumerate all of them and map them to user readable messages.

Reply:
I still believe for the HTTP error cases blanket rules might work well enough. We know when we get 400/403 errors and we can report them aptly. Other errors like a 500 can be reported as an invalid config combination error. Would leave it to @aniketshrimal' to break the ice here.

Reply:
I'll add another layer of validation in the next iteration. Can we merge this in to make the devel cut for now. The next one will go along with the system/integration tests and will be available for review?

Reply:
Sounds good, Raul. I agree with what @mukkachaitanya is saying. Let's try to add our own error-messages in case of standard http response code - 400/403 in the next iteration.

Reply:
Also let us not try to cover all the cases in one function. It is true this function is slightly overloaded as we are trying to throw validation error for all the response cases except one (400 Bad Request - successful validation case). We should try to reduce the scope of this function and just capture specific validation errors. The left out cases can be mapped to error-mapping later whenever the exception occurs. Overall, we should try to cleanup this function in the next iteration.

Reply:
The cons are that while with this approach, for 3 instances: there are 3 calls in sequence and for another separate call per path, it would double to 6 calls sequentially. I was hoping to avoid that. I'll add the other version in the next review and we can go with that approach and see how it works out.